### DISCUSSION TITLE
Default export for modular styles

### DISCUSSION THREAD

**Discussion by wmertens** [vanilla-extract-css/vanilla-extract#924] [Ideas]
V-E imports the generated CSS as .css files. I changed the plugin so that if there's a default export, it gets replaced with the generated CSS and no import is added.

This is great for Qwik, so that a style is only added to the HTML if it is used.

Is this good for a standard feature?

https://github.com/wmertens/styled-vanilla-extract/blob/main/README.md#only-emitting-styles-you-use

**Comment by n8sabes** [comment]
That seems pretty cool. As **1)** Qwik continues to gain traction and **2)** the community continues to lean toward vanilla-extract, and **3)** one of the fundamental premises of both VE and Qwik is about optimizing (minimizing) the load time while remaining pure, it seems to be a worthwhile optimization contribution as you described it. At least I would want that in my projects.

As for any technical objections or architectural impacts, it would be good to hear from @mattcompiles or other core contributors.
